### DISCUSSION TITLE
Environment variables (or .env files)

### DISCUSSION THREAD

**Discussion by JJ** [community/community#26541] [Actions]
<p>I see that Github actions support global environment variables at the job level right now. However, suppose those variables are in a .env file. Would there be some way to dump that into GITHUB_ENV file so that it can be used as an environment variable there?</p>

**Comment by airtower-luna** [comment]
<p>Assuming your <code>.env</code> file follows <a href="https://docs.github.com/en/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable">the format required by <code>GITHUB_ENV</code></a> you can just append it all:</p>
<pre><code class="lang-yaml">steps:
  - name: read environment file
    run: cat .env &gt;&gt;${GITHUB_ENV}
</code></pre>
<p>If the <code>.env</code> file uses a different format you’ll have to run it through some kind of converter tool instead of simply copying, e.g. a script that reads the environment file and writes to <code>GITHUB_ENV</code> in the expected format.</p>

**Comment by JJ** [comment]
<p>Sorry, I somehow didn’t put the key word here, at the job level I mean. You can’t <code>run</code> anything at that level, right?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You mean <em>before</em> any steps get run? There’s no option for that. Maybe because the <code>.env</code> file wouldn’t be available before checkout anyway?</p>
<p>On the other hand settings in <code>GITHUB_ENV</code> remain for the rest of the job, so the effect is very similar.</p>
